decam_process.py

```diff
@@ -15,161 +15,82 @@
-    usual post-ISR data (bkgd, calexp, icExp, icSrc, postISR).
-
-    diffIm - Do difference imaging with a visit as a template and
-    one or more as science.
-    USAGE:
-    $ python hits_ingest.py diffIm
-    BASH EQUIVALENT:
-    $ imageDifference.py processedrepo --id visit=sciencevisit ccdnum=ccdnum
-            --templateId visit=templatevisit --output diffimrepo
-            -C diffim_config.py
-    ** to run from bash, 'diffim_config.py' must exist and contain, e.g.,
-        from lsst.ip.diffim.getTemplate import GetCalexpAsTemplateTask
-        config.getTemplate.retarget(GetCalexpAsTemplateTask)
-        config.detection.thresholdValue=5.0
-        config.doDecorrelation=True
-    RESULT:
-    diffimrepo/deepDiff/v+sciencevisit populated with difference images
-    and catalogs of detected sources (diaSrc, diffexp, and metadata files)
-                        '''))
-    parser.add_argument('-f', '--files',
-                        help='Input files or directory for ingest or ingestCalibs.')
+    parser.add_argument('-d', '--dataset',
+                        help="Location on disk of dataset_root, which contains subdirectories of raw data, calibs, etc.")
+    parser.add_argument('-o', '--output',
+                        help="Location on disk where output repos will live.")
+    # TODO: implement this argument
+    # parser.add_argument('-i', '--id',
+    #                    help="String containing visit and ccd information. Typically set as 'visit=12345, ccd=5'.")
     args = parser.parse_args()
-    if (args.task == 'ingest' or args.task == 'ingestCalibs') and not args.files:
-        raise IOError('-f is required to ingest images or calibrations.')
-    if args.files is None:
-        datadir = None
-        datafiles = None
-    else:
-        datadir = args.files
-        if os.path.isdir(datadir):
-            datafiles = glob(os.path.join(datadir, '*.fz'))
-        else:
-            datafiles = glob(datadir)
-
-    # Run whichever task has been requested
-    if args.task == 'ingest':
-        doIngest(repo, datafiles)
-    elif args.task == 'ingestCalibs':
-        doIngestCalibs(repo, calibrepo, datafiles)
-    elif args.task == 'processCcd':
-        doProcessCcd(repo, calibrepo, processedrepo, visit, ccdnum)
-    elif args.task == 'diffIm':
-        doDiffIm(processedrepo, sciencevisit, ccdnum, templatevisit, diffimrepo)
+
+    if not os.path.isdir(args.output):
+        os.mkdir(args.output)
+    repo = os.path.join(args.output, 'ingested')
+    calibrepo = os.path.join(args.output, 'calibingested')
+    processedrepo = os.path.join(args.output, 'processed')
+    diffimrepo = os.path.join(args.output, 'diffim')
+    types = ('*.fits', '*.fz')
+    datafiles = []
+    allcalibdatafiles = []
+    for files in types:
+        datafiles.extend(glob(os.path.join(args.dataset, 'raw', files)))
+        allcalibdatafiles.extend(glob(os.path.join(args.dataset, 'calib', files)))
+
+    # Ignore wtmaps and illumcors
```

Comment:
Fair enough! Mucking with `main()` is a bit lower on my priority list since `ap_verify` doesn't care about it, but you're both correct that this function is rather lengthy with multiple functionalities. I'll see what I can do.